specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -2,199 +2,242 @@
-* NW verifies the app token against AppSID extracted from the channel URL
-* NW sends the Post request to WNS’s NN which contains the persistent socket with the client
-* WNS’s NN maps the channel URL to the correct deviceId by decoding the deviceId from the channel URL and sends an NFY to the correct client.
-* WPN client receives the NFY and processes the channelId in the payload to retrieve the correct AppId from the WPN database.
-* WPN client routes the notification to the appropriate app. If it needs to interface with the App, it uses BI, callbacks or WNF depending on the App Registration. If the payload is a toast, it gets directly routed to the Shell.
+* [The Windows Push Notification Services (WNS) Overview](https://docs.microsoft.com/windows/uwp/design/shell/tiles-and-notifications/windows-push-notification-services--wns--overview)
+
+![Legacy Flow For Reunion](Legacy.png)
 
 ## The problems today
 
-**Portal Registration**: The entire Registration flow for Push is dependent on registering an app in the partner centre portal by [associating an app with the Windows Store](https://docs.microsoft.com/en-us/azure/notification-hubs/notification-hubs-windows-store-dotnet-get-started-wns-push-notification). As a result, non-store provisioned apps like Win32 cannot avail of this feature even if they want to.
+**Portal Registration**: The entire Registration flow for Push is dependent on registering an app in the <br/>
+partner center portal by [associating an app with the Windows Store](https://docs.microsoft.com/azure/notification-hubs/notification-hubs-windows-store-dotnet-get-started-wns-push-notification). As a result, non-store provisioned <br/>
+apps like Win32 cannot avail of this feature even if they want to.
 
-**Client Registration**: Unpackaged 3rd party applications do not have a way today to register with the client because the API is private. The only way to get an implicit client side registration is to package the app as an MSIX or SSP which may not satisfy 3rd party requirements.
+**Client Registration**: Unpackaged 3rd party applications do not have a way today to register with the <br/>
+client because the API is private. The only way to get an implicit client side registration is to package <br/>
+the app as an MSIX Desktop Bridge or SSP which may not satisfy 3rd party requirements.
 
-**Push Channels**: All the Push APIs are dependent on the caller app having an identity at the minimal. However, even packaged Win32 apps that have an identity cannot use our Public APIs to subscribe to Push Channels because they may not be store provisioned. Sideloaded UWPs that aren't store distributed have exactly the same problem.
+**Push Channels**: All the Push APIs are dependent on the caller app having an identity at the minimal. <br/>
+However, even packaged Win32 apps that have an identity cannot use our Public APIs to subscribe to Push <br/>
```

Comment:
can be simpler. "This means Win32 and packaged apps not distributed through the store can't use this system"